libs/sdk-ui-dashboard/src/model/commandHandlers/render/renderingWorker.ts

```diff
@@ -120,6 +128,14 @@ function* collectAsyncRenderTasks(config: RenderingWorkerConfiguration): SagaIte
         // New async render requested, spawn it.
         const asyncRenderTask = yield spawn(waitForAsyncRenderResolution, asyncRenderId, config);
         asyncRenderTasks.set(asyncRenderId, asyncRenderTask);
+
+        // expected count of async renders was received, no need to wait longer
+        if (
+            config.asyncRenderExpectedCount !== undefined &&
+            asyncRenderTasks.size >= config.asyncRenderExpectedCount
```

Comment:
What about custom widget registrations? Could this be resolved "too soon" for dashboard with plugins and custom widgets (that are not part of the layout)?

Reply:
The expected count is sent there only when we are sure we know the expected count => see the condition here https://github.com/gooddata/gooddata-ui-sdk/pull/4756/files#diff-368a1b4b95f0e1665739046d06742402b5500a94886bd8bc5139187279d56c58R112-R113 we just want to make sure we resolve non-plugin dashboards asap, and just have the increased timeout when there are plugins and we don't know the count.